### PR TITLE
Hangzhi

### PR DESCRIPTION
**Pull request by Alina-PANG**
basic setups for the backend of public feeds page:
1. datastore changed
2. new servlet

### CODE REVIEW

File: .factorypath [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<factorypath>
```
Comment (kiatchuan) [inline .factorypath]: I believe this is an Eclipse-specific IDE file, could you remove this from the pull request and add it to the gitignore?

**Comment by Alina-PANG** [comment]
@Mengyang, I have moved all the scripts to the footer part of the html file. You will find them there :)

File: .factorypath (hunk 1 above)
Comment (Alina-PANG) [reply .factorypath]: ok just fixed..

**Comment by LiMengyang990726** [comment]
> @LiMengyang990726 , I have moved all the scripts to the footer part of the html file. You will find them there :)

I believe that this part:
```
function createNavBar(){
  document.getElementById('nav-bar').innerHTML = 
  '<nav class="navbar navbar-expand-lg navbar-dark bg-dark">\
  <a class="navbar-brand" href="#">Team 35</a>\
  <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav"\
    aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">\
    <span class="navbar-toggler-icon"></span>\
  </button>\
  <div class="collapse navbar-collapse" id="navbarNav">\
    <ul class="navbar-nav" id="navigation">\
      <li class="nav-item">\
        <a class="nav-link" href="./index.html">Home</a>\
      </li>\
      <li class="nav-item ">\
        <a class="nav-link" href="./aboutus.html">Our Team </a>\
      </li>\
      <li class="nav-item ">\
        <a class="nav-link" href="./feed.html">Public Feeds</a>\
      </li>\
      <li class="nav-item ">\
        <a class="nav-link" href="./stats.html">Statistics</a>\
      </li>\
    </ul>\
  </div>\
</nav>';
} 
```
is better to be changed to the following:
```
if (loginStatus.isLoggedIn) {
        navigationElement.appendChild(createListItem(createLink('/user-page.html?user=' + loginStatus.username, 'Your Page')));

        navigationElement.appendChild(createListItem(createLink('/stats.html','Statistics')));

        // more links for public feed etc. can be added here
        
        navigationElement.appendChild(
          createListItem(createLink('/logout', 'Logout')));
      } else {
        navigationElement.appendChild(
          createListItem(createLink('/login', 'Login')));
      }
```
As the later one is more efficient and is able to add login credentials.

**Comment by Alina-PANG** [comment]
Sure we can make that change.. i'm trying to include all the nav and button stuff in the js. But definitely can remove the content inside '<ul>' and add in your code after calling the createNavBar function

**Comment by LiMengyang990726** [comment]
> Sure we can make that change.. i'm trying to include all the nav and button stuff in the js. But definitely can remove the content inside ul tag and add in your code after calling the createNavBar function

Sure! That's also a good way of doing that!
I think seperating the links in another function is a good practice, but the current one may not be as good as the original syntax as doing `createListItem(createLink)`

**Pull request merged**